### ISSUE TITLE
Hot key reading bug?

### ISSUE THREAD

**Issue by kivdev** [boppreh/keyboard#244]
```
import keyboard,time
while True:
    hotkey = keyboard.read_hotkey()
    print(hotkey)
    time.sleep(0.5)
```

I press:
ctrl+z
ctrl+x
ctrl+c
ctrl+v
ctrl+b
ctrl+shit+b

Program output:
ctrl+z
ctrl+x+z
ctrl+c+x+z
ctrl+v+x+z
ctrl+b+x+z
ctrl+shift+B+x+z

Why do some keys stick?
I encountered the same situation when I was tracking keystrokes with GetKeyState. Ordinary keys have become like CapsLock and the like.

**Comment by roninpawn** [comment]
Same here!
Seeing as its been months, did you ever workaround it @kivdev ?

**Comment by roninpawn** [comment]
Here's my solution.

    def getHotKey():
        newkey = keyboard.read_key()
        hotkey = lastkey = newkey
        while keyboard.is_modifier(newkey):
            newkey = keyboard.read_key()
            if newkey != lastkey:
                hotkey += f"+{newkey}"
                lastkey = newkey
        return hotkey`

Got to say... I don't find this a particularly useful library. The in-built obfuscation of key_codes, replacing them with their common, readable associations means more work for all my implementations.

There's a numpad 3 and there's a keyboard 3, and btw, the numpad 3 could also be the page down command if numlock isn't on. This library makes me dance for its amusement before I can check if a specific one of these has been pressed without having all of them return true. I guess if you WANT all of them to return true then this is an asset. But without methods to just access the keycodes directly, its more limiting than helpful in half of all uses.

Or /is/ there a read_key command that returns 29+42+56+46 instead of "ctrl+shift+alt+C"?

**Comment by roninpawn** [comment]
Here's a complete non-blocking solution for capturing a hotkey after a pressing a button in tkinter. It falls out of the wait cycle if you click anywhere or the window loses focus.
 
```
import tkinter as tk
import keyboard
from time import sleep

class UI(tk.Tk):
    def __init__(self):
        super().__init__()

        self.button = tk.Button(self, command=self.getKey, text="Capture hotkey")
        self.button.pack()

    def getKey(self):
        def cancel(event=None):
            nonlocal done
            self.button.unbind("<Button-1>", click_id)
            self.button.unbind("<FocusOut>", focus_id)
            keyboard.unhook(key_hook)
            done = "Cancel"

        def do_it(event):
            nonlocal lastkey, hotkey, done
            if event.event_type == keyboard.KEY_UP:
                done = True
                return
            elif lastkey != event.name:
                hotkey[0].append(event.name)
                hotkey[1].append(event.scan_code)
                lastkey = event.name

        done = False
        lastkey = ""
        hotkey = [[], []]

        key_hook = keyboard.hook(do_it, True)
        click_id = self.button.bind_all("<Button-1>", cancel)
        focus_id = self.button.bind_all("<FocusOut>", cancel)

        self.button.config(text="PRESS NOW")
        while not done:
            self.update()
            sleep(.015)
        if done != "Cancel": cancel()
        self.button.config(text="Capture hotkey")
        print(hotkey)


window = UI()
window.mainloop()
```

The answer to my prior complaints against and questions for this library are: yes, no, kind of, and almost. When you get down to it, this library just wants you to instantiate one of its hook-methods to monitor keystrokes; then work with the data in the KeyboardEvent class fed to the callback on an event. Which is fair enough. The KeyboardEvent class contains everything you could possibly want or need to accomplish any key-input analysis.

That said, if you're trying to just reach in and grab a current-state of the keyboard without having hooked in first, this library just isn't built to accommodate that. You can get the current scan_codes of any key that's down, but not the key_names. And without both halves of that dataset, you can't REALLY be sure what key is being pressed unless its a single-duty key.

The tldr is: Just use keyboard.hook(), process the events yourself, and don't ask questions.

**Comment by xoiga123** [comment]
```python
import keyboard,time
while True:
    hotkey = keyboard.read_hotkey()
    keyboard.stash_state()
    print(hotkey)
    time.sleep(0.5)
```

Basically just put `keyboard.stash_state()` between multiple `keyboard.read_hotkey()`, this works for me.

**Comment by MatteP99** [comment]
You can simply pass **suppress=False** while you're calling the function. At least, it works for me on both Linux and Windows.
Like this:
`𝘷𝘢𝘭 = keyboard.read_hotkey(suppress=False)`
